barman/recovery_executor.py

```diff
@@ -1807,13 +1810,327 @@ def _backup_copy(self, backup_info, dest, remote_command=None, **kwargs):
+
+        # Perform the standard recovery process passing the synthetic backup
+        return super(IncrementalRecoveryExecutor, self).recover(
+            synthetic_backup_info, dest, remote_command=remote_command, **kwargs
+        )
+
+    def _combine_backups(self, backup_info, dest):
+        """
+        Combines the backup chain into a single synthetic backup using the
+        pg_combinebackup utility.
+
+        :param barman.infofile.LocalBackupInfo backup_info: an incremental
+            backup in the chain
+        :param str dest: the directory where the synthetic backup is going
+            to be mounted on
+        :return barman.infofile.SyntheticBackupInfo: the backup info file of the
+            combined backup
+        """
+        self.combine_start_time = datetime.datetime.now()
+
+        # Build the synthetic backup info from the incremental backup as it has
+        # the most recent data relevant to the recovery. Also, the combine process
+        # should be transparent to the end user so e.g. the .barman-recover.info file
+        # that is created on destination and also the backup_id that is appended to the
+        # manifest file in further steps of the recovery should be the same as the incremental
+        synthetic_backup_info = SyntheticBackupInfo(
+            self.server,
+            base_directory=dest,
+            backup_id=backup_info.backup_id,
+        )
+        synthetic_backup_info.load(filename=backup_info.filename)
+
+        dest_dirs = [synthetic_backup_info.get_data_directory()]
+
+        # Maps the tablespaces from the old backup directory to the new synthetic
+        # backup directory. This mapping is passed to the pg_combinebackup as input
+        tbs_map = {}
+        if backup_info.tablespaces:
+            for tablespace in backup_info.tablespaces:
+                source = backup_info.get_data_directory(tablespace_oid=tablespace.oid)
+                destination = synthetic_backup_info.get_data_directory(
+                    tablespace_oid=tablespace.oid
+                )
+                tbs_map[source] = destination
+                dest_dirs.append(destination)
+
+        # Prepare the destination directories for pgdata and tablespaces
+        for d in dest_dirs:
+            self._prepare_destination(d)
+
+        # Retrieve pg_combinebackup version information
+        remote_status = self._fetch_remote_status()
+
+        # Get the backup chain data paths to be passed to the pg_combinebackup
+        backups_chain = self._get_backup_chain_path(backup_info)
+
+        self._start_message(synthetic_backup_info)
+
+        pg_combinebackup = PgCombineBackup(
+            destination=synthetic_backup_info.get_data_directory(),
+            command=remote_status["pg_combinebackup_path"],
+            version=remote_status["pg_combinebackup_version"],
+            app_name=self.config.streaming_backup_name,
```

Comment:
```suggestion
            app_name=None,
```

I would avoid reusing that application name as I guess that can sound confusing to the user if they inspect `pg_stat_replication`. I suggested `None` because that's what `PgVerifyBackup` class does.

If we want to have `application_name` in the connection string, I would say to use something else other than the streaming backup name.